packages/components/src/components/domainproperties/dataset/models.ts

```diff
@@ -79,33 +81,41 @@ export class DatasetModel extends Record({
             return new DatasetModel({ ...newDataset, domain });
         } else {
             const domain = DomainDesign.create(raw.domainDesign);
-            return new DatasetModel({ ...raw.datasetDesign, domain });
+            return new DatasetModel({ ...raw.options, domain });
         }
     }
 
     hasValidProperties(): boolean {
         let isValidKeySetting = true;
 
         if (this.getDataRowSetting() === 2) {
-            isValidKeySetting = this.keyPropertyId !== undefined && this.keyPropertyId !== 0;
+            isValidKeySetting = this.keyPropertyName !== undefined && this.keyPropertyName !== '';
         }
 
-        return this.name !== undefined && this.name !== null && this.name.trim().length > 0 && isValidKeySetting;
+        return (
+            this.name !== undefined &&
+            this.name !== null &&
+            this.name.trim().length > 0 &&
+            this.label !== undefined &&
+            this.label !== null &&
+            this.label.trim().length > 0 &&
+            isValidKeySetting
+        );
     }
 
     isNew(): boolean {
-        return !this.entityId;
+        return !this.datasetId;
```

Comment:
I thought we had switched this to be entityId on purpose to handle the case where the user is creating a dataset and puts in a dataset id in the advanced dialog and then goes back to the dialog. With this using datasetId, doesn't that scenario result in a disabled input for the dataset id even though the dataset hasn't been created yet?

Reply:
Should it be added?

Reply:
yes, for the scenario you mentioned. @RosalineP can you look into sending the entity id back in response?

Reply:
done